parser/wasm_parse/mod.rs

```diff
@@ -242,77 +334,97 @@ impl<'a> Parse<'a> for wasmparser::ModuleReader<'a> {
+                wasmparser::Payload::TableSection(mut reader) => {
+                    reader.parse_edges(items, ())?;
                 }
-                wasmparser::SectionCode::Memory => {
-                    section
-                        .get_memory_section_reader()?
-                        .parse_edges(items, ())?;
+                wasmparser::Payload::MemorySection(mut reader) => {
+                    reader.parse_edges(items, ())?;
                 }
-                wasmparser::SectionCode::Global => {
-                    section
-                        .get_global_section_reader()?
-                        .parse_edges(items, ())?;
+                wasmparser::Payload::GlobalSection(mut reader) => {
+                    reader.parse_edges(items, ())?;
                 }
-                wasmparser::SectionCode::Export => {
-                    section
-                        .get_export_section_reader()?
-                        .parse_edges(items, (&indices, idx))?;
+                wasmparser::Payload::ExportSection(mut reader) => {
+                    reader.parse_edges(items, (&indices, idx))?;
                 }
-                wasmparser::SectionCode::Start => {
-                    StartSection(section).parse_edges(items, (&indices, idx))?;
+                wasmparser::Payload::StartSection { func, range } => {
+                    StartSection {
+                        function_index: func,
+                        data: &self.data[range.start..range.end],
+                    }
+                    .parse_edges(items, (&indices, idx))?;
                 }
-                wasmparser::SectionCode::Element => {
-                    section
-                        .get_element_section_reader()?
-                        .parse_edges(items, (&indices, idx))?;
+                wasmparser::Payload::ElementSection(mut reader) => {
+                    reader.parse_edges(items, (&indices, idx))?;
                 }
-                wasmparser::SectionCode::Data => {
-                    section.get_data_section_reader()?.parse_edges(items, ())?;
+                wasmparser::Payload::DataSection(mut reader) => {
+                    reader.parse_edges(items, ())?;
                 }
-                wasmparser::SectionCode::DataCount => {
-                    DataCountSection(section).parse_edges(items, ())?;
+                wasmparser::Payload::DataCountSection { range, .. } => {
+                    DataCountSection {
+                        size: range.end - range.start,
+                    }
+                    .parse_edges(items, ())?;
                 }
-                wasmparser::SectionCode::Code | wasmparser::SectionCode::Function => {
+                wasmparser::Payload::CodeSectionStart { .. }
+                | wasmparser::Payload::FunctionSection { .. } => {
                     unreachable!("unexpected code or function section found");
                 }
+
+                _ => {} // @@@
```

Comment:
Let's replace this with the explicit things that we have left that are unhandled. Should it also be ignored, or point to the unreachable above?